### PR TITLE
Update dependencies

### LINKED ISSUE #2011: Use BOMs for the dependencies that distribute BOMs
`netty-bom`, `jackson-bom` and maybe some other dependencies.

Need to make sure our POMs (including `armeria-bom` and all other artifacts) are generated correctly while working on this.

### PR DESCRIPTION
**Pull request by ikhoon**
* Add new BOMs
  * grpc: 1.23.0 -> grpc-bom:1.24.0
  * io.dropwizard.metrics: 4.0.0 ->  metrics-bom:4.1.0
  * jetty: 9.4.20.v20190813 -> jetty-bom:9.4.21.v20190926
  * junit -> junit-bom:5.5.2 (no upgrade)
* Compile
  * jackson-bom: 2.9.9.20190807 -> 2.10.0
  * java-jwt: 3.8.2 -> 3.8.3
  * gax-grpc: 1.48.1 -> 1.49.0
  * micrometer: 1.2.1 -> 1.3.0
* Test
  * mockito: 3.0.0 -> 3.1.0
* Closes #2011

### CODE REVIEW

File: dependencies.yml [hunk 1]
```diff
@@ -3,10 +3,14 @@
...
 boms:
-  - com.fasterxml.jackson:jackson-bom:2.9.9.20190807
+  - com.fasterxml.jackson:jackson-bom:2.10.0
   - io.zipkin.brave:brave-bom:5.7.0
   # NOTE: When changing this, re-evaluate netty-tcnative-boringssl-static below
   - io.netty:netty-bom:4.1.42.Final
+  - io.grpc:grpc-bom:1.24.0
+  - io.dropwizard.metrics:metrics-bom:4.1.0
+  - org.eclipse.jetty:jetty-bom:9.4.21.v20190926
+  - org.junit:junit-bom:5.5.2
```
Comment (trustin) [inline dependencies.yml]: Please sort.

File: dependencies.yml [hunk 2]
```diff
@@ -233,21 +225,13 @@ junit:
 
 org.junit.jupiter:
   junit-jupiter-api:
-    version: &JUNIT_JUPITER_VERSION '5.5.2'
     javadocs:
     - https://junit.org/junit5/docs/5.5.1/api/
```
Comment (trustin) [inline dependencies.yml]: `5.5.2/api`

Comment (ikhoon) [reply dependencies.yml]: Oops, Thanks!

File: dependencies.yml (hunk 1 above)
Comment (ikhoon) [reply dependencies.yml]: Sure! I sorted it alphabetically.

File: dependencies.yml [hunk 3]
```diff
@@ -3,10 +3,14 @@
 #
 
 boms:
-  - com.fasterxml.jackson:jackson-bom:2.9.9.20190807
-  - io.zipkin.brave:brave-bom:5.7.0
+  - com.fasterxml.jackson:jackson-bom:2.10.0
+  - io.dropwizard.metrics:metrics-bom:4.1.0
   # NOTE: When changing this, re-evaluate netty-tcnative-boringssl-static below
   - io.netty:netty-bom:4.1.42.Final
+  - io.grpc:grpc-bom:1.24.0
```
Comment (minwoox) [inline dependencies.yml]: This should be before netty?

**Pull request merged**